google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/instance/AppEngineServerModel.java

```diff
@@ -162,20 +165,28 @@ public void checkConfiguration() throws RuntimeConfigurationException {
           "App Engine facet not found in '" + artifact.getName() + "' artifact");
     }
 
-    if (CloudSdkService.getInstance().getSdkHomePath() == null) {
+    CloudSdkService cloudSdkService = CloudSdkService.getInstance();
+    if (cloudSdkService.getSdkHomePath() == null
+          || StringUtils.isEmpty(cloudSdkService.getSdkHomePath().toString())) {
```

Comment:
An empty SDK is invalid, and that is going to be caught when you call sdk.validateCloudSdk() shortly after.

I'm considering an empty SDK different from a null SDK. For an SDK to be empty, the user has to have made it that way (previously, it's null, hence unspecified). For me, an empty string falls under "invalid string provided by user", so the messaging should be coherent.

Also, there could be a file system where an empty string is a valid location.